3_harmonize/src/clean_wqp_data.R

```diff
@@ -3,29 +3,103 @@
 #' @description 
 #' Function to harmonize WQP data in preparation for further analysis. Included
 #' in this function are steps to unite diverse characteristic names by assigning
-#' them to more commonly-used water quality parameter names. 
+#' them to more commonly-used water quality parameter names, and to flag missing
+#' records as well as duplicate records.
 #' 
+#' @param wqp_data data frame containing the data downloaded from the WQP, 
+#' where each row represents a unique data record.
 #' @param char_names_crosswalk data frame containing columns "char_name" and 
 #' "parameter". The column "char_name" contains character strings representing 
 #' known WQP characteristic names associated with each parameter.
-#' @param wqp_params list object where each element of the list is a named vector.
-#' The vectors correspond with a parameter group and contain character string(s)
-#' representing known WQP characteristic names associated with each parameter. 
+#' @param commenttext_missing character string(s) indicating which strings from
+#' the WQP column "ResultCommentText" correspond with missing result values. By 
+#' default, the column "ResultCommentText" will be searched for the following 
+#' strings: "analysis lost", "not analyzed", "not recorded", "not collected", 
+#' and "no measurement taken", but other values may be added as well. 
```

Comment:
Just feel concerned that same people may not realize they need to write `c('analysis lost', 'not analyzed', 'not recorded', 'not collected', 'no measurement taken', 'new value', 'second new value')` but maybe that is not something we should guard against since it is just how R functions and their default args work.

```suggestion
#' and "no measurement taken", but other values may be added by passing in a new
#' vector with all values to be treated as missing. 
```

Reply:
Good point! These arguments get kinda long and so I like to hide them when I can using default args.